examples/gross_pitaevskii_custom_potential.jl

```diff
@@ -0,0 +1,114 @@
+# # Gross-Pitaevskii equation in one dimension with forces and custom potential
+
+# We solve the 1D Gross-Pitaevskii equation with a custom potential.
+# This is similar to the
+# previous example ([Gross-Pitaevskii equation in one dimension](@ref)) where
+# computing forces was not possible as we used an `ExternalFromReal` term
+# instead of an `AtomicLocal` term (see [`Model` datastructure](@ref)).
+# Here, we use `ElementCustomIonPotential`
+# with a custom potential to generate an `AtomicLocal` term for which we can
+# compute forces.
+using DFTK
+using LinearAlgebra
+
+# ## The model
+# First we set up the lattice. For a 1D case we supply two zero lattice vectors,
+a = 10
+lattice = a .* [[1 0 0.]; [0 0 0]; [0 0 0]];
+
+# In this example, we want to generate two Gaussian potentials generated by ions
+# localized at positions ``x_1`` and ``x_2``, that are expressed in ``[0,1)`` in
+# fractional coordinates. ``|x_1 - x_2|`` should be different from ``0.5`` to
+# break symmetry and get nonzero forces.
+x1 = 0.2
+x2 = 0.8;
+# We define the width of the Gaussian potential generated by one ion
+L = 0.5;
+# We set the potential in its real and Fourier forms
+pot_real(x) = exp(-(x/L)^2)
+pot_fourier(q::T) where {T <: Real} = exp(- (q*L)^2 / 4);
+# And finally we build the elements and set their positions in the `atoms` array
+ion = ElementCustomIonPotential(1, pot_real, pot_fourier) # 1 is the charge of each ion
+atoms = [ion => [x1*[1,0,0], x2*[1,0,0]]];
+
+# Parameters for the nonlinear potential ``α C ρ^{α-1}``
```

Comment:
these comments were already done in the GP example, you can remove them so that the reader knows to focus on the new stuff